### PR TITLE
Fix #3075: View() doesn't support 2D vectors that aren't matrices

### LINKED ISSUE #3075: View() doesn't support 2D vectors that aren't matrices
RTVS silently produces no output View(x)  even though x can be coerced to a data.frame
this is related to Issue #3000 where View is identified as a workaround

source of Issue:
```
library(quantmod)
View(getSymbols("SPY", auto.assign = FALSE))
```

`tail(as.data.frame(getSymbols("SPY", auto.assign = FALSE)))`

produces:

```
           SPY.Open SPY.High SPY.Low SPY.Close SPY.Volume SPY.AdjFactor
2017-01-18   226.54   226.80  225.90    226.75   51146400             1
2017-01-19   226.84   227.00  225.41    225.91   65357500             1
2017-01-20   226.70   227.31  225.97    226.74  116431700             1
2017-01-23   226.74   226.81  225.27    226.15   72331900             1
2017-01-24   226.40   228.08  226.27    227.60   89144800             1
2017-01-25   228.70   229.57  228.51    229.57   80932500             1
```

Also, this works as expected:

`View(as.data.frame(getSymbols("SPY", auto.assign = FALSE)))`

### PR DESCRIPTION
**Pull request by int19h**
Combine all the various grid viewers into one to simply logic and make it possible to do checks in a strictly defined order.

Remove most class-based checks, and look at typeof, length and dimenstions instead.

Support grid data fetching for tibble and similar objects.

Add viewer and grid data tests for various missing cases.

### CODE REVIEW

File: src/Package/Test/DataInspect/ViewersTest.cs [hunk 1]
```diff
@@ -72,19 +76,19 @@ public class ViewersTest : IAsyncLifetime {
...
-                gridViewer.Should().NotBeNull().And.BeOfType<Viewer1D>();
+                gridViewer.Should().NotBeNull().And.BeOfType<GridViewer>();
 
                 gridViewer = await _aggregator.GetViewer(session, REnvironments.GlobalEnv, "list(c(1:10))");
-                gridViewer.Should().NotBeNull().And.BeOfType<ListViewer>();
+                gridViewer.Should().NotBeNull().And.BeOfType<GridViewer>();
 
                 gridViewer = await _aggregator.GetViewer(session, REnvironments.GlobalEnv, "c(1:10)");
-                gridViewer.Should().NotBeNull().And.BeOfType<VectorViewer>();
+                gridViewer.Should().NotBeNull().And.BeOfType<GridViewer>();
```
Comment (AlexanderSher) [inline src/Package/Test/DataInspect/ViewersTest.cs]: I believe you don't need to call `.NotBeNull()` - `BeOfType` checks for null.

Comment (int19h) [inline src/Package/Test/DataInspect/ViewersTest.cs]: Yep. Removed.

File: src/Package/Impl/DataInspect/Viewers/GridViewer.cs [hunk 2]
```diff
@@ -2,27 +2,37 @@
...
             base(aggregator, evaluator) { }
 
         #region IObjectDetailsViewer
-        public override bool CanView(IRValueInfo evaluation) {
-            if (evaluation != null && evaluation.Classes.Any(t => _tableClasses.Contains(t))) {
-                return evaluation.Dim != null && evaluation.Dim.Count == 2;
+
+        public override bool CanView(IRValueInfo value) {
+            // We can only view collections that have elements.
+            if (value == null || value.Length == null || value.Length == 0) {
```
Comment (AlexanderSher) [inline src/Package/Impl/DataInspect/Viewers/GridViewer.cs]: `evaluation?.Length == null || evaluation.Length == 0`

**Comment by AlexanderSher** [comment]
👍

**Pull request merged**